src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -7144,6 +7164,81 @@ def _update_artifact_type(params):
                    "Possible values: war, jar, ear, zip, startup, script, static", params.artifact_type)
 
 
+def _get_instances_list_arm_internal(params):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    try:
+        client = web_client_factory(params.cmd.cli_ctx)
+        sub_id = get_subscription_id(params.cmd.cli_ctx)
+        if params.slot:
+            base_url = (
+                f"subscriptions/{sub_id}/resourceGroups/{params.resource_group_name}/providers/Microsoft.Web/sites/"
+                f"{params.webapp_name}/slots/{params.slot}/instances"
+                f"?api-version={client.DEFAULT_API_VERSION}"
+            )
+        else:
+            base_url = (
+                f"subscriptions/{sub_id}/resourceGroups/{params.resource_group_name}/providers/Microsoft.Web/sites/"
+                f"{params.webapp_name}/instances?api-version={client.DEFAULT_API_VERSION}"
+            )
+
+        url = params.cmd.cli_ctx.cloud.endpoints.resource_manager + base_url
+        response = send_raw_request(params.cmd.cli_ctx, "GET", url)
+        if response.status_code == 200:
+            return response.json().get("value", [])
+        return None
+    except Exception as ex:
+        logger.info("Getting exception while trying to get instances list: %s", ex)
+
+
+def _get_instance_id_internal(param):
```

Comment:
since i only need one instance Id, i think we can combine this login in the above method itself and return a single value from it